code/core/src/manager/components/sidebar/useFavicon.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-env browser */
+import { useEffect, useRef } from 'react';
+
+const STATUSES = ['active', 'critical', 'negative', 'positive', 'warning'] as const;
+
+let initialIcon: string | undefined;
+
+const setFavicon = (link: HTMLLinkElement, suffix: string = '') => {
+  initialIcon ??= link.href || `./favicon.svg`;
+
+  const href = initialIcon.replace(/([^/]+)(\.\w+)([?#].*)?$/, `$1${suffix}$2$3`);
+  const img = new Image();
+  img.onload = () => (link.href = href);
+  img.onerror = () => (link.href = initialIcon!);
+  img.src = href;
```

Comment:
is this essentially _"try to load the favicon file with the correct suffix, if it fails then revert back"_?

sounds like it will produce _a lot_ of annoying 404 message in the Network and Console DevTools, I think that should be avoided, we should find another way.

Also, won't this cause the user's favicon to be replaced with our own with a status dot if the user has called their favicon `favicon.svg`?